orbstation/code/antagonists/contract_killer.dm

```diff
@@ -0,0 +1,223 @@
+/// A re-imagining of the "Obsessed" antag, sharing some of its mechanics and goals but not the unpleasant flavor.
+/datum/antagonist/contract_killer
+	name = "Contract Killer"
+	show_in_antagpanel = TRUE
+	antagpanel_category = "Other"
+	job_rank = ROLE_CONTRACT_KILLER
+	antag_hud_name = "contractkiller"
+	ui_name = "AntagInfoMinor"
+	show_name_in_check_antagonists = TRUE
+	roundend_category = "contract killers"
+	count_against_dynamic_roll_chance = FALSE
+	silent = FALSE
+	suicide_cry = "FOR THE PAY!! Wait-"
+	preview_outfit = /datum/outfit/contract_killer
+
+/datum/antagonist/contract_killer/on_gain()
+	forge_objectives()
+	return ..()
+
+/datum/antagonist/contract_killer/greet()
+	owner.current.playsound_local(get_turf(owner.current), 'orbstation/sound/antag_alerts/contractkilleralert.ogg', 100, FALSE, pressure_affected = FALSE, use_reverb = FALSE)
+	owner.announce_objectives()
+
+/datum/antagonist/contract_killer/forge_objectives()
+	var/list/objectives_left = list("intel", "polaroid")
+	var/datum/objective/assassinate/kill_objective = new
+	kill_objective.owner = owner
+	var/datum/mind/target_mind = kill_objective.find_target()
+	var/obj/family_heirloom
+
+	//if the target has an heirloom, the contract killer might try to steal it
+	var/datum/quirk/item_quirk/family_heirloom/heirloom_quirk = locate() in target_mind.current.quirks
+	if(heirloom_quirk)
+		family_heirloom = heirloom_quirk.heirloom?.resolve()
+		if(family_heirloom)
+			objectives_left += "heirloom"
+
+	// If the target is not the captain, someone in the same department might also be a kill objective.
+	if(!is_captain_job(target_mind.assigned_role))
+		objectives_left += "coverup"
+
+	for(var/i in 1 to 2)
+		var/chosen_objective = pick(objectives_left)
+		objectives_left.Remove(chosen_objective)
+		switch(chosen_objective)
+			if("intel")
```

Comment:
To be honest these strings should _probably_ be file-scope defines just to minimise the chance of spelling error when making changes or future additions